docs/getting-started/advanced-guide/schema.md

```diff
@@ -0,0 +1,430 @@
+
+Schemas are often separated into two JSON files (`elements.json` & `types.json`), but it's worth remembering that your schema can be broken up further if required.
+
+For example if you have a Graph that is made up of 5 different data sources, you may find it easier to develop and maintain by splitting your schema into multiple parts, for example 5 Element Schemas and 1 Types schema.
+Alternatively, a single overall schema file could be used for Elements and Types. There's also no requirement to have a single Types schemas.
+When you construct your Gaffer graph you must provide all the Schema parts. If provided separately, these will then be merged together to form an internal schema used for the Graph.
+
+When using Java directly, you don't need to use JSON files to store you schemas - although this is still recommended. See the [Java API](#java-api) section for more about interacting with Schemas in Java.
+
+The sections below walkthrough the features of Schemas in detail and explain how to accurately model your data in Gaffer.
+
+
+## Elements schema
+The Elements schema is designed to be a high level document describing what information your Graph contains, i.e. the different kinds of edges and entities and the list of properties associated with each.
+Essentially this part of the schema should just be a list of all the entities and edges in the graph. 
+Edges describe the relationship between a source vertex and a destination vertex. 
+Entities describe a vertex. We use the term "element" to mean either and edge or and entity.
+
+When defining an element we must provide a "group". This is a unique string identifier for each element.
+Groups must be completely unique and cannot be shared between edges and entities.
+
+### Edges
+Edges must have the following:
+
+- `source` - Type of object to use as the source vertex in your graph. Needs to be a type defined in the [Types schema](#types).
+- `destination` - Type of object to use as the destination vertex in your graph. Can either be the same type as `source`, or a different type.
+- `directed` - Tells Gaffer if the edge is directed or undirected. Needs to be a type which means true or false, see [Types](#true--false) for more info.
+
+When an Edge is undirected in Gaffer (`directed` is `false`), it is treated as if the relationship was bidirectional, and the vertices of the edge do not have an authoritative source and destination.
+Thus, the undirected edges `A -- B` and `B -- A` are equal, and will therefore be aggregated together.
+Gaffer will present the undirected edges Vertices in natural ordering, so a client will also see the above edge presented as `A, B`.
+It will be aggregated with any other edge that has the same source and destination and is also undirected.
+For example an undirected Edge that is added, `A -> B`, will be aggregated with another Edge of `B -> A` if they are undirected.
```

Comment:
I feel like this paragraph is a bit repetitive? Instead I would combine a couple of sentences:
Thus, the undirected edges A -- B and B -- A are equal, and will therefore be aggregated with any other edge that has the same source and destination and is also undirected. Gaffer will present the undirected edges vertices in natural ordering, so a client will also see the above edge presented as A, B. Therefore an undirected Edge that is added, A -- B, will be aggregated with another Edge of B -- A if they are undirected.

Reply:
I also wasn't sure if the two bottom two examples should be -- instead of -> as they are meant to be undirected

Reply:
This section could certainly be more concise, I'll reword.

Reply:
I've make it more like what's suggested and also trimmed it further. That last sentence is probably not needed but it does make things very clear.